operator/src/main/resources/scripts/introspectDomain.py

```diff
@@ -554,6 +608,108 @@ def validateServerTemplateNapListenPortIsSet(self, server_or_template):
+    return leasingRequired
+
+  def validateJTAMigrationPolicy(self, server_or_template, cluster, isServerTemplate=False):
+    leasingRequired = False
+    migrationPolicy = self.getJTAMigratableTargetMigrationPolicy(server_or_template, isServerTemplate)
+    # possible values are "manual", "exactly-once", and "failure-recovery".
+    # All values except "manual" requires cluster leasing.
+    if migrationPolicy is not None and migrationPolicy != "manual":
+      leasingRequired = True
+      if self.isConsensusLeasing(cluster):
+        if isServerTemplate:
+          type_str = "server template "
+        else:
+          type_str = "server "
+        self.addConsensusLeasingError("JTA migratable target in " + type_str + self.name(server_or_template)
+                                      + " in cluster " + self.name(cluster) + " requires automatic service migration")
+    return leasingRequired
+
+  def addMigrationError(self, message):
+    self.addError(message + " Set SKIP_LEASING_VALIDATIONS environment variable to 'True' to skip this validation.")
+
+  def addConsensusLeasingError(self, message):
+    self.addError(message + ", but cluster is configured to use consensus leasing which is not " +
+                  "supported by the WebLogic Kubernetes Operator. " +
+                  "Please configure cluster to use database leasing " +
+                  "or set SKIP_LEASING_VALIDATIONS environment variable to 'True' to skip this validation.")
+
+  def findMigratableTarget(self, name):
+    for migratableTarget in self.env.getDomain().getMigratableTargets():
+      if migratableTarget.getName() == name:
+        return migratableTarget
+    return None
+
+  def validateMigratableTargets(self, cluster):
+    leasingRequired = False;
+    for migratableTarget in self.env.getDomain().getMigratableTargets():
+      if migratableTarget.getCluster() is cluster:
+        # possible migration policy values are "manual", "exactly-once", and "failure-recovery".
+        # All values except "manual" requires cluster leasing.
+        migrationPolicy = migratableTarget.getMigrationPolicy()
+        if migrationPolicy is not None and migrationPolicy != "manual":
+          # check JMS resources targeted to this migratable target
+          leasingRequired |= self.validateJMSResourcesLeasing(migratableTarget)
+    return leasingRequired
+
+  def validateJMSResourcesLeasing(self, migratableTarget):
+    leasingRequired = False;
+    domain = self.env.getDomain()
+    for messagingBridge in domain.getMessagingBridges():
+      leasingRequired |= self.validateJMSResourceMigrationPolicy(messagingBridge, migratableTarget, "Messaging bridge")
+    for jdbcStore in domain.getJDBCStores():
+      leasingRequired |= self.validateJMSResourceMigrationPolicy(jdbcStore, migratableTarget, "JDBCStore")
+    for fileStore in domain.getFileStores():
+      leasingRequired |= self.validateJMSResourceMigrationPolicy(fileStore, migratableTarget, "FileStore")
```

Comment:
Add JMS Servers, SAF Agents, and Path Services to this list.  Technically this isn't needed since their respective stores will have the same target, but it wouldn't hurt just in case it's possible to configure one without a store.

Reply:
Sure. added validations for JMS servers, SAF agents, and path services